tasksheets/sheet1.md

```diff
@@ -0,0 +1,53 @@
+Task 1
+======
+We use Rust. For details on the infrastructure, refer to the README.
+
+
+Task 2
+======
+Done.
+
+
+Task 3
+======
+
+(1) MiniJava Program
+----------------
+```java
+class Foo {
+    public Foo[] bar;
+
+    public static void main(String[] args) {
+        System.out.println(42);
+    }
+}
+```
+
+(2) On the Turing-Completeness of MiniJava
+--------------------------------------
+MiniJava is turing complete, since it provides both conditional expressions (*IfStatement*),
+as well as an iteration construct (*WhileStatement*). Unlimited memory is achieved via the
+32-bit `int` type as a local variable and using recursion (*Method* and *MethodInvocation*)
+to "allocate" more memory. Since there is no mechanism for getting the address of a variable
+(like e.g. `&varname` in C), the size of the Stack is not bounded, and can grow arbitrarily
```

Comment:
Took me a few seconds to understand why this implication holds...

Reply:
Reading it again, my formulation is not even true. The implication only holds in the opposite direction (addressable memory => bounded memory). There could be other "features" of the language that I missed which impose bounds on the memory... Proving turing-completeness is really hard ;-) I'll try to reformulate and be a little more verbose